src/routes/api/use-chat-tools/+server.ts

```diff
@@ -0,0 +1,44 @@
+import { createOpenAI } from '@ai-sdk/openai';
+import { convertToCoreMessages, streamText } from 'ai';
+import { z } from 'zod';
+import { env } from '$env/dynamic/private';
+import type { RequestHandler } from './$types';
+
+const groq = createOpenAI({
+  baseURL: 'https://api.groq.com/openai/v1',
+  apiKey: env.GROQ_API_KEY ?? '',
+});
+
+export const POST = (async ({ request }) => {
+  const { messages } = await request.json();
+
+  const result = await streamText({
+    model: groq('llama-3.1-8b-instant'),
+    messages: convertToCoreMessages(messages),
+    tools: {
+      getWeatherInformation: {
```

Comment:
?

Reply:
All these were present in the reference repo. We have just updated the Chat bot section for now. I will clean all these in next commit.